File: Golang/Find All Numbers Disappeared in an Array.go
//1
func findDisappearedNumbers(nums []int) []int {
    var res []int
    for i:=0;i<len(nums);i++{
        temp := abs(nums[i]) - 1
        if nums[temp] > 0{
            nums[temp] *= -1
        }
    }

    for i, v := range nums{
        if v > 0 {
            res = append(res , i+1)
        }
    }
    return res
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

//2
func findDisappearedNumbers(nums []int) []int {
    length := len(nums)
    if length == 0 { return nil }
    res := make([]int, length)
    for _, v := range nums {
        res[v-1] = v
    }
    fmt.Print(res)
    j := 0
    for i := 0; i < length; i++ {
        if res[i] == 0 { 
            res[j] = i + 1
            j++
        }
    } 
    return res[:j]
}

File: Python3/Find All Numbers Disappeared in an Array.py
#1
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)):
            temp = abs(nums[i]) - 1
            if nums[temp] > 0:
                nums[temp] *= -1
        
        res = []
        for i,n in enumerate(nums):
            if n > 0:
                res.append(i+1)
        
        return res

#2 use set
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        return set(i for i in range(1, len(nums) + 1)) - set(nums)